Scholars Portal Journals.js

```diff
@@ -0,0 +1,197 @@
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) { 
+	var myArticles = doc.getElementById('my-articles');
+	if (myArticles) {
+  		Zotero.monitorDOMChanges(myArticles.parentElement, {attributes: true, attributeFilter: ["style"]});
+	}
+	// see if saved list is toggled open
+	if (doc.getElementsByClassName('inner-wrap-open') [0]) {
+		if (getItems(doc, true)) {
+			return 'multiple'
+		}
+	} else if (url.indexOf("/search?q") != -1 || url.indexOf("/browse/") != -1) {
+		if (getItems(doc, true)) {
+			return 'multiple'
+		}
+	} else if (url.indexOf("/details/") != -1) {
+		return 'journalArticle'
+	} 
+}
+
+function doWeb(doc, url) {
+	var type = detectWeb(doc, url);
+	if(type == "multiple") {
+		var list = getItems(doc, url);
+		Zotero.selectItems(list, function(selectedItems) {
+			if(!selectedItems) return true;
+			var articles = [];
+			for(var i in selectedItems) {
+				var article = 'http://'+doc.domain+'/ris?uri='+i
+				articles.push(article);
+			}
+			ZU.doGet(articles, scrape);
+		})
+	} else {
+		var uri = getURI(url);
+		var article = 'http://'+doc.domain+'/ris?uri='+uri
+		ZU.doGet(article, scrape);
+	}
+}
+
+function getURI(url){
+	if(url.indexOf("xml") != -1){
+		var a = url.indexOf("details");
+		var b = url.indexOf("xml");
+		uri = url.substring(a+7,b+3);
+		return uri
+	} else if(url.indexOf("resolver.scholarsportal.info/resolve/") != -1)  {
+		uri = url.split("resolver.scholarsportal.info/resolve")[1] + ".xml";
+		return uri
+	}
+}
+
+function getItems(doc, url) {
+	var items = {}, found = false;
+	if (doc.getElementsByClassName('inner-wrap-open') [0]) {
+		var titles = ZU.xpath(doc.getElementById('my-articles-list'), './/div[@class = "title"]/h3/a');
+		for (var i=0; i<titles.length; i++) { 
+			var title = ZU.trimInternal(titles[i].textContent);
+			var uri = getURI(titles[i].href);
+			items[uri] = title;
+			found = true;
+		}
+	} else {
+		if(doc.URL.indexOf("/browse") != -1){
+			var titles = ZU.xpath(doc, './/div/h4/a');
+		}
+		else {
+			var titles = ZU.xpath(doc.getElementById('result-list'), './/div[@class = "details"]/h3/a');
+		
+		}
+		for (var i=0; i<titles.length; i++) { 
+			var title = ZU.trimInternal(titles[i].textContent);
+			var uri = getURI(titles[i].href);
+			items[uri] = title;
+			found = true;
+		}
+	}
+	return found ? items : false;
+}
+
+function scrape(text, doc) { 
+	// loading RIS transformer. 
+	var translator = Zotero.loadTranslator("import");
+	translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+	translator.setString(text);
+	translator.setHandler("itemDone", function(obj, item) {
+		var uri = getURI(item.attachments[0].path);
+		var pdfURL = "/pdf" + uri;
+		item.url = "http://journals.scholarsportal.info/details" + uri;
+		item.attachments = [{}];
```

Comment:
you're adding an empty attachment here. You can either remove the curly brackets (i.e. the empty object) here, or, perhaps better, delete this line and, instead of pushing to the attachments array do

```
item.attachments = [{
		url: pdfURL,
		title: "Scholars Portal Full Text PDF",
		mimeType: "application/pdf"
	}]
```